src/content/app/tools/blast/components/blast-input-sequences/BlastInputSequences.tsx

```diff
@@ -35,8 +41,24 @@ const BlastInputSequences = () => {
     appendEmptyInputBox,
     setUncommittedSequencePresence
   } = useBlastForm();
+
   const shouldAppendEmptyInput = useAppSelector(getEmptyInputVisibility);
 
+  const blastFormContext = useContext(BlastFormContext);
+
+  const previousSequences = usePrevious(sequences);
+
+  useEffect(() => {
+    if (previousSequences?.length !== sequences.length) {
+      blastFormContext?.clearSequenceValidityFlags();
```

Comment:
We don't need to clear it here. We will be anyways resetting it on L58 below.

Reply:
Good point... i have removed this line now.